synapse/handlers/sync.py

```diff
@@ -1020,15 +1021,16 @@ def _generate_sync_entry_for_groups(self, sync_result_builder):
 
     @measure_func("_generate_sync_entry_for_device_list")
     @defer.inlineCallbacks
-    def _generate_sync_entry_for_device_list(self, sync_result_builder,
+    def _generate_sync_entry_for_device_list(self, sync_result_builder, user_id,
```

Comment:
the `user_id` param is redundant, and immediately overwritten on the first line of the function.